src/backend/distributed/metadata/node_metadata.c

```diff
@@ -910,11 +1155,24 @@ ActivateNode(char *nodeName, int nodePort)
 						BoolGetDatum(true));
 	}
 
-	SetUpDistributedTableDependencies(workerNode);
+	SetUpDistributedTableWithDependencies(workerNode);
```

Comment:
as we discussed earlier in person there are several confusion points around here:
1)  try moving `SyncNodeMetadataToNode` above
2) Move `LocalGroupIdUpdateCommand`  to right after SyncNodeMetadataToNode
3) In general, `SetUpDistributedTableWithDependencies` touches both shell tables and all distributed tables. And, `ResyncMetadataCommandList` deals with pg_dist_part/shard.

One very useful refactoring would be to unify `SyncCitusTableMetadata` -- my suggestion in another place -- and this place. What we need to do is simply iterate over all the propagated tables, and generate the same commands. So, maybe `SyncCitusTableMetadataCommandList` function which creates the commands. On create_dist_table, we only use the list for 1 table, here we use it for all.

 Such a refactor would increase the readability substantially, making it much better than what we have.

And, all non-table related stuff can go to their own functions.  

Otherwise, it'd be very hard for anyone to follow the `SetUpDistributedTableWithDependencies` vs `ResyncMetadataCommandList` in a meaningful way

What do you think?